certification/internal/policy/operator/deployable_by_olm.go

```diff
@@ -178,30 +182,37 @@ func (p *DeployableByOlmCheck) installedCSV(operatorData OperatorData, openshift
 	defer cancel()
```

Comment:
The defer cancel() should be inside the goroutine. And just ctx := context.Background()

Reply:
I believe the defer statement should be the immediate LoC after the context creation. Otherwise, if spinning the goroutine fails for any reason, the `cancel()` method will never be called.

Reply:
That's why you add the Timeout to the context within the goroutine. The goroutine should not expect a fully configured context. It should not make the assumption that the context has the timeout. You explicitly add the timeout to the context within the goroutine. Which means, outside of the goroutine, where you create the context, you are just creating it as context.Background(). The caller should not be making the decision about how that context will be further used. And I think that's why you're seeing the delay. If you defer cancel() within the goroutine (which you would not have if you are initializing as ctx := context.Background()), then it runs that when the goroutine finishes, not when the outer function returns.

Reply:
Done.